c/tskit/tables.h

```diff
@@ -697,28 +697,21 @@ typedef struct {
 /* Leave room for more positive check flags */
 #define TSK_NO_CHECK_POPULATION_REFS (1 << 12)
 
-/* Flags for load tables */
-#define TSK_BUILD_INDEXES (1 << 0)
+/* Flags for table collection load or init
+ * As flags are passed though from load to init they share a namespace */
+#define TSK_LOAD_INIT_NO_METADATA (1 << 0)
```

Comment:
LOAD_INIT seems a bit weird. Aren't these properties of the table collection/tables, which we're specifying at init time? So we could say something like

```
// This table has no metadata
TSK_TABLE_NO_METADATA (1 << 0)

// This table collection will have no reference sequence/tables/edge metadata
TSK_TC_NO_REFERENCE_SEQUENCE(1 << 0)
TSK_TC_NO_TABLES(1 << 1)
TSK_TC_NO_EDGE_METADATA (1 << 2)
```